src/edu/csus/ecs/pc2/clics/API202306/CLICSScoreboard.java

```diff
@@ -0,0 +1,129 @@
+ *
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CLICSScoreboard {
+
+    @JsonProperty
+    private String time;
+
+    @JsonProperty
+    private String contest_time;
+
+    @JsonProperty
+    private CLICSContestState state;
+
+    @JsonProperty
+    private CLICSScoreboardRow [] rows;
+
+    /**
+     * Fill in the scoreboard information
+     *
+     */
+    public CLICSScoreboard(IInternalContest model, IInternalController controller, Group group, Integer division)  throws IllegalContestState, JAXBException, IOException {
+
+        DefaultScoringAlgorithm scoringAlgorithm = new DefaultScoringAlgorithm();
+
+        Properties properties = ScoreboardUtilities.getScoringProperties(model);
+
+        ArrayList<Group> groupList = null;
+
+        if(group != null) {
+            groupList = new ArrayList<Group>();
+            groupList.add(group);
+        }
+        // legacy - standings are created as XML, and we convert that to JSON.
+        String xml = scoringAlgorithm.getStandings(model, null, division, groupList, properties, StaticLog.getLog());
+
+        ContestStandings contestStandings = ScoreboardUtilities.createContestStandings(xml);
+
+        // This is what we want to return:
+        //        {
+        //            "time": "2014-06-25T14:13:07.832+01",
+        //            "contest_time": "4:13:07.832",
+        //            "state": {
+        //              "started": "2014-06-25T10:00:00+01",
+        //              "ended": null,
+        //              "frozen": "2014-06-25T14:00:00+01",
+        //              "thawed": null,
+        //              "finalized": null,
+        //              "end_of_updates": null
+        //            },
+        //            "rows": [
+        //              {"rank":1,"team_id":"123","score":{"num_solved":3,"total_time":340},"problems":[
+        //                {"problem_id":"1","num_judged":3,"num_pending":1,"solved":false},
+        //                {"problem_id":"2","num_judged":1,"num_pending":0,"solved":true,"time":20},
+        //                {"problem_id":"3","num_judged":2,"num_pending":0,"solved":true,"time":55},
+        //                {"problem_id":"4","num_judged":0,"num_pending":0,"solved":false},
+        //                {"problem_id":"5","num_judged":3,"num_pending":0,"solved":true,"time":205}
+        //              ]}
+        //            ]
+        //          }
+        time = ZonedDateTime.now( ZoneOffset.UTC ).format( DateTimeFormatter.ISO_INSTANT);
+        contest_time = model.getContestTime().getElapsedTimeStr();
+        state = new CLICSContestState(model, null);
+
+        ArrayList<CLICSScoreboardRow>rowsArray = new ArrayList<CLICSScoreboardRow>();
+        HashMap<String, String> probEleToShort = new HashMap<String, String>();
+
+        // create a mapping of each problem's element ID to its shortname.
+        // we will use shortname as the problem id in the problem list for each team's solutions
```

Comment:
Using `shortname` for the problem id seems to be a good idea.  I recall that there was some _other_ file (unfortunately I don't recall which one) where problem id's were defined using the problem `ElementId`.  (I think I commented on that at the time of review.)   I wish I could recall where that was so I could request it be fixed :)

Reply:
Well, `CLICSProblem `uses the short name for the `problemId `on the event feed and API endpoint, so...